core/storage/trie/impl/trie_storage_backend_impl.hpp

```diff
@@ -8,13 +8,19 @@
 
 #include "common/buffer.hpp"
 #include "outcome/outcome.hpp"
+#include "storage/spaced_storage.hpp"
 #include "storage/trie/trie_storage_backend.hpp"
 
 namespace kagome::storage::trie {
 
-  class TrieStorageBackendImpl : public TrieStorageBackend {
+  class TrieStorageBackendImpl : public TrieNodeStorageBackend,
+                                 public TrieValueStorageBackend {
```

Comment:
These are just interfaces.

Reply:
These are ambigous/conflicting interfaces.
Without `virtual` inheritance this class will contain `BufferStorage` instance twice (for each non-virtual inheriting subclass).

Reply:
> `BufferStorage` is abstract base class and wont be instantiated.

You can't create `BufferStorage` variable because it's vtable is empty,
but subclass always contains/stores superclass (`struct SubClass { SuperClass superclass; }`).